src/vpn-worker.ts

```diff
@@ -102,68 +103,67 @@ const worker = async (instanceId: number, serviceId: number) => {
 		}
 	});
 
-	let draining = false;
-	const drainConnections = async () => {
+	const drainConnections = _.once(async () => {
```

Comment:
Given that this piece of code plays a critical role, could we have a small comment here describing what the function does and why? More specifically - are there any caveats and/or assumptions (such as any k8s settings) that our future selves should be reminded of?

Reply:
Going by the FD thread, it is clear that quite some thought & discussion has gone into it.